### PR TITLE
Fix editing and saving of configs on the Config Management page

### LINKED ISSUE #360: Can't edit existing configs using edit form on /configs page
The create/edit form shows an error growler when trying to save:
Screen Shot 2024-06-10 at 2 06 15 PM

### PR DESCRIPTION
**Pull request by rgallettonf**
closes #360

### CODE REVIEW

File: projects/ziti-console-lib/src/lib/features/wrappers/zac-wrapper.service.ts [hunk 1]
```diff
@@ -812,6 +813,26 @@ export class ZacWrapperService extends ZacWrapperServiceClass {
         });
     }
 
+    redefineObject(obj) {
```
Comment (dovholuknf) [inline projects/ziti-console-lib/src/lib/features/wrappers/zac-wrapper.service.ts:816]: `redefineObject` seems like a strange name to me. it feels more like a `resetObject` to me, but maybe that's my weak js-foo?

File: projects/ziti-console-lib/src/lib/features/wrappers/zac-wrapper.service.ts [hunk 2]
```diff
@@ -812,6 +813,26 @@ export class ZacWrapperService extends ZacWrapperServiceClass {
...
+            } else {
+                if (typeof obj[prop] === "string" && obj[prop].trim().length==0) {
+                    delete obj[prop];
+                } else {
+                    if (typeof obj[prop] === "object") {
+                        obj[prop] = this.redefineObject(obj[prop]);
+                        if (Object.keys(obj[prop]).length==0) {
+                            delete obj[prop];
+                        }
+                    }
```
Comment (dovholuknf) [inline projects/ziti-console-lib/src/lib/features/wrappers/zac-wrapper.service.ts:829]: there's no 'else' here. feels like a console.log or ... anything ? would be good here?

Comment (dovholuknf) [reply projects/ziti-console-lib/src/lib/features/wrappers/zac-wrapper.service.ts:829]: are numbers covered? or just left the same?

Comment (rgallettonf) [reply projects/ziti-console-lib/src/lib/features/wrappers/zac-wrapper.service.ts:829]: > are numbers covered? or just left the same?

Should be left the same

**Review by dovholuknf — APPROVED** [review]

**Pull request merged**